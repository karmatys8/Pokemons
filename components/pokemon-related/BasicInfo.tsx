import React from "react";

import { ThemedView } from "../ThemedView";
import { ThemedText } from "../ThemedText";

import { styles } from "@/assets/styles";

type InfoProps = {
  name: string;
  height: number;
  weight: number;
};

const BasicInfo: React.FC<InfoProps> = ({ name, height, weight }) => (
  <ThemedView style={styles.stepContainer}>
    <ThemedText type="subtitle">{`Name: ${name}`}</ThemedText>
    <ThemedText>{`Height: ${height}`}</ThemedText>
    <ThemedText>{`Weight: ${weight}`}</ThemedText>
  </ThemedView>
);

export default BasicInfo;
