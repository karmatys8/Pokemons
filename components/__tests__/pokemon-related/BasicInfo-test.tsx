import React from "react";
import renderer from "react-test-renderer";
import BasicInfo from "@/components/pokemon-related/BasicInfo";

test("renders correctly", () => {
  const tree = renderer
    .create(<BasicInfo name="nidoking" height={14} weight={620} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
