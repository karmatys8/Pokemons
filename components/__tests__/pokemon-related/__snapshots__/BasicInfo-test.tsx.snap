// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "gap": 8,
        "marginBottom": 8,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        undefined,
        undefined,
        undefined,
        {
          "fontSize": 20,
          "fontWeight": "bold",
        },
        undefined,
        undefined,
      ]
    }
  >
    Name: nidoking
  </Text>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    Height: 14
  </Text>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    Weight: 620
  </Text>
</View>
`;
